src/commands/dev/gcs/server.rs

```diff
@@ -0,0 +1,133 @@
+use crate::commands::dev::gcs::headers::{destructure_response, structure_request};
+use crate::commands::dev::ServerConfig;
+use crate::terminal::emoji;
+
+use std::sync::{Arc, Mutex};
+
+use chrono::prelude::*;
+
+use hyper::client::{HttpConnector, ResponseFuture};
+use hyper::header::{HeaderName, HeaderValue};
+use hyper::http::uri::InvalidUri;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Client as HyperClient, Request, Response, Server, Uri};
+
+use hyper_tls::HttpsConnector;
+
+const PREVIEW_HOST: &str = "rawhttp.cloudflareworkers.com";
+
+/// performs all logic that takes an incoming request
+/// and routes it to the Workers runtime preview service
+pub(super) async fn serve(
+    server_config: ServerConfig,
+    preview_id: Arc<Mutex<String>>,
+) -> Result<(), failure::Error> {
+    // set up https client to connect to the preview service
+    let https = HttpsConnector::new();
+    let client = HyperClient::builder().build::<_, Body>(https);
+
+    let listening_address = server_config.listening_address.clone();
+
+    // create a closure that hyper will use later to handle HTTP requests
+    let make_service = make_service_fn(move |_| {
+        let client = client.to_owned();
+        let server_config = server_config.to_owned();
+        let preview_id = preview_id.to_owned();
+        async move {
```

Comment:
mebbe not here, but i'd suggest pulling these nested anonymous functions into named functions for clarity. not 100% necessary tho.

Reply:
I actually spent like an hour trying to figure out how to do this but couldn't because of the `move` statements. Happy for some help on this one